File: GeeksForGeeks/Easy/Reverse_coding.cpp
#include <iostream>
using namespace std;

#define endl '\n'

int main()
{
    int test_case;
    ios::sync_with_stdio(false);

    cin >> test_case;
    int input, given_output;
    while (test_case--)
    {
        cin >> input >> given_output;
        if (input * (input+1) /2 == given_output) {
            cout << "1" << endl;
        }
        else {
          cout << "0" << endl;
        }
    }
}

File: GeeksForGeeks/Easy/Reverse_coding.java
import java.util.Scanner;

class GFG
 {
	public static void main (String[] args)
	{
	    Scanner scan = new Scanner(System.in);
	    int testcase = scan.nextInt();

      int input, givenOutput;
    	while(testcase!=0)
    	{
          input       = scan.nextInt();
          givenOutput = scan.nextInt();
          if ((input * (input + 1))/2 == givenOutput) {
              System.out.println(1);
          }
    	    else {
              System.out.println(0);
          }
          --testcase;
    	}
    	scan.close();
	}
}
